python/foglamp/services/core/service_registry/monitor.py

```diff
@@ -31,60 +30,61 @@ class Monitor(object):
     _DEFAULT_PING_TIMEOUT = 1
     """Timeout for a response from any given micro-service"""
 
-    _logger = None  # type: logging.Logger
+    _DEFAULT_MAX_ATTEMPTS = 15
 
-    def __init__(self):
+    _logger = None
 
-        self._logger = logger.setup("SMNTR", level=logging.INFO)
+    def __init__(self):
+        self._logger = logger.setup(__name__, level=20)
 
         self._monitor_loop_task = None  # type: asyncio.Task
         """Task for :meth:`_monitor_loop`, to ensure it has finished"""
         self._sleep_interval = None  # type: int
         """The time (in seconds) to sleep between health checks"""
         self._ping_timeout = None  # type: int
         """Timeout for a response from any given micro-service"""
+        self._max_attempts = None  # type: int
+        """Number of max attempts for finding a heartbeat of service"""
 
     async def _sleep(self, sleep_time):
         await asyncio.sleep(sleep_time)
 
     async def _monitor_loop(self):
         """async Monitor loop to monitor registered services"""
         # check health of all micro-services every N seconds
-        _MAX_ATTEMPTS = 15
-        """Number of max attempts for finding a heartbeat of service"""
 
         while True:
+            self._logger.info("Starting next round of service monitoring, sleep/i:{} ping/t:{} max/a:{}".format(
+                self._sleep_interval, self._ping_timeout, self._max_attempts))
             for service_record in ServiceRegistry.all():
-                attempt_count = 1
-                """Number of current attempt to ping url"""
-
-                url = "{}://{}:{}/foglamp/service/ping".format(service_record._protocol, service_record._address,
-                                                               service_record._management_port)
-                async with aiohttp.ClientSession() as session:
-                    while attempt_count < _MAX_ATTEMPTS + 1:
-                        # self._logger.info("Attempting service %s with try count %d", url, attempt_count)
-                        try:
+                # No need to try ping if service status is either Unregistered or Failed
+                if service_record._status in [ServiceRecord.Status.Running, ServiceRecord.Status.Doubtful]:
```

Comment:
If we also ping failed services we will allow them to recover if there is a temporary issue.